src/plugins/plugin.gridlines.js

```diff
@@ -0,0 +1,166 @@
+'use strict';
+
+module.exports = function(Chart) {
+	var helpers = Chart.helpers;
+	var globalDefaults = Chart.defaults.global;
+
+	// Stores all gridLines that should be displayed
+	var lines = [];
```

Comment:
A plugin is potentially called for many charts, so it's not safe to assume that the `update` and `draw` will be called synchronously per chart (especially if there is animations). This `lines` array would contain lines for different charts.